src/contract/fungible.rs

```diff
@@ -435,4 +519,116 @@ mod test {
             .collect::<HashSet<_>>();
         assert_eq!(generators.len(), 1);
     }
+
+    #[test]
+    #[should_panic]
```

Comment:
Tests with `should_panic` attribute may be imprecise, because the test could pass even if the test panics for a different reason from the one we were expecting. So I would remove the attribute and invert the `assert!` check (expecting false instead of true) or add the `expected` parameter to the `should_panic` attribute (in this case `#[should_panic(expected = "assertion failed: secp256k1_zkp::verify_commitments_sum_to_equal(SECP256K1, &[a, b], &[c, d])")]`)

Reply:
Great point, reversed conditions